lib/conjur/command/variables.rb

```diff
@@ -124,4 +151,56 @@ class Conjur::Command::Variables < Conjur::Command
     end
 
   end
+
+  def self.prompt_for_id
+    highline.ask('Enter the id: ')
+  end
+
+  def self.prompt_for_group
+    highline.ask('Enter the group: ', ->(name) { @group && @group.roleid } ) do |q|
+      q.validate = ->(name) do
+        name.empty? || (@group = api.group(name)).exists?
+      end
+      q.responses[:not_valid] = "Group '<%= @answer %>' doesn't exist, or you don't have permission to use it"
+    end
+  end
+
+  def self.prompt_for_kind
+    highline.ask('Enter the kind: ') {|q| q.default = @default_kind }
+  end
+
+  def self.prompt_for_mime_type
+    highline.ask('Enter the MIME type: ') {|q| q.default = @default_mime_type }
+  end
+
+  def self.prompt_for_annotations
+    highline.say('Add annotations (press enter to finish):')
+    {}.tap do |annotations|
+      until (name = highline.ask('annotation name: ')).empty?
+        annotations[name] = read_till_eof('annotation value (^D to finish):')
+      end
+    end
+  end
+
+  def self.prompt_for_value
+    read_till_eof('Enter the value (^D on its own line to finish):')
+  end
+  
+  def self.highline
+    require 'highline'
+    @highline ||= HighLine.new($stdin,$stderr)
+  end
+
+  def self.read_till_eof(prompt = nil)
+    highline.say(prompt) if prompt
+    [].tap do |lines|
```

Comment:
For all I/O, or just read_till_eof? 

For all I/O, HighLine seemed liked it offered some benefits over puts/gets.

For read_till_eof, it seemed worthwhile to keep it consistent with the rest of the code.